src/v/kafka/security/acl.h

```diff
@@ -0,0 +1,447 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+#include "kafka/types.h"
+#include "model/fundamental.h"
+#include "seastarx.h"
+#include "utils/named_type.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <absl/container/btree_map.h>
+#include <absl/container/flat_hash_set.h>
+#include <fmt/core.h>
+
+namespace kafka {
+
+// cluster is a resource type and the acl data model requires that resources
+// have names, so this is a fixed name for that resource.
+using acl_cluster_name = named_type<ss::sstring, struct acl_cluster_name_type>;
```

Comment:
an interesting idea would be to understand how big on average the acl names are. and perhaps build a new `sstring` basic string type with a short string optimization of say N chars, where N is some small and known name value (i.e: we can enforce in cloud to be like 32 chars). (just a thought)

Reply:
I am wondering if this can even be named type based on `std::string_view` then we can make it `constexpr` ?

Reply:
I don’t see how, since it has to store the state; string-view only works for things that are only string literals.

Reply:
it looks like this is always constant.

Reply:
seems useful if we can really guess well. another option might be to use something like a sorted string table and do some sort of arena allocation. this'll be a non issue until we start seeing large numbers of users / rules.